ultrasonic/src/main/kotlin/org/moire/ultrasonic/view/SongView.kt

```diff
@@ -0,0 +1,395 @@
+/*
+ This file is part of Subsonic.
+
+ Subsonic is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Subsonic is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with Subsonic.  If not, see <http://www.gnu.org/licenses/>.
+
+ Copyright 2009 (C) Sindre Mehus
+ */
+package org.moire.ultrasonic.view
+
+import android.content.Context
+import android.graphics.drawable.AnimationDrawable
+import android.graphics.drawable.Drawable
+import android.text.TextUtils
+import android.view.LayoutInflater
+import android.widget.Checkable
+import org.koin.java.KoinJavaComponent.get
+import org.koin.java.KoinJavaComponent.inject
+import org.moire.ultrasonic.R
+import org.moire.ultrasonic.data.ActiveServerProvider.Companion.isOffline
+import org.moire.ultrasonic.domain.MusicDirectory
+import org.moire.ultrasonic.featureflags.Feature
+import org.moire.ultrasonic.featureflags.FeatureStorage
+import org.moire.ultrasonic.service.DownloadFile
+import org.moire.ultrasonic.service.MediaPlayerController
+import org.moire.ultrasonic.service.MusicServiceFactory.getMusicService
+import org.moire.ultrasonic.util.Util
+import org.moire.ultrasonic.util.VideoPlayerType
+import org.moire.ultrasonic.view.EntryAdapter.SongViewHolder
+import timber.log.Timber
+
+/**
+ * Used to display songs in a `ListView`.
+ *
+ * @author Sindre Mehus
```

Comment:
You can remove this, the code is yours and it is not necessary to quote the author of deleted code. If you want you can add your name :wink:

Reply:
Oh okay! I've removed the author as I strongly believe creating this code is a team effort.

Reply:
Yes, is a team effort, but in new code cannot put other name as author because is _a lie_. The original authors are always credited in git history and in contributors list.